pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -374,3 +386,12 @@ def _save_loggers_on_train_batch_end(self) -> None:
         should_flush_logs = self.trainer.logger_connector.should_flush_logs
         if should_flush_logs and self.trainer.is_global_zero and self.trainer.logger is not None:
             self.trainer.logger.save()
+
+    def on_save_checkpoint(self) -> Dict:
+        out = super().on_save_checkpoint()
+        out["val_loop_done"] = self.val_loop.done
+        return out
+
+    def on_load_checkpoint(self, state_dict: Dict) -> None:
+        self._val_loop_done = state_dict["val_loop_done"]
+        super().on_load_checkpoint(state_dict)
```

Comment:
Does the epoch loop really need to handle the information of its subloop? can we keep it contained?

Reply:
Good catch ! Removed entirely.